test/spec/groupCtrlSpec.js

```diff
@@ -149,6 +149,20 @@ describe("Autocomplete controller", function() {
     });
   });
 
+  describe("performCompletion", function() {
+    it('auto complete triggers', function() {
+      scope.autoComplete = function(v) {}; // this function normally will be filled in by the auto-complete directive
```

Comment:
You can just do `scope.autoComplete = sinon.spy()`

I've refactored the test to look like this:

```js
it('triggers autoComplete', function() {
  scope.autoComplete = sinon.spy();

  var msg = {user: "boo"}; // scope.autoComplete only cares about user
  scope.query = {text: "b"};
  scope.performCompletion(msg);

  expect(scope.query).to.be.eq(null);
  expect(scope.autoComplete.callCount).to.be.eq(1);
  expect(scope.autoComplete).to.have.been.calledWith(msg);
});
```

Whenever possible, it's best to think of the describe and it block as two parts of a sentence, so "performComplete triggers autoComplete" makes more sense than "performCompletion auto complete triggers" (unless Yoda is writing the test, then I guess I can be more lenient :wink:)

I try to keep comments spare in it blocks. The test should be descriptive. In this case, I added the comment for the msg variable because it's a truncated version of the object it's representing, and that's okay because autocomplete only cares about the user.